janusgraph-core/src/main/java/org/janusgraph/graphdb/management/ConfigurationManagementGraph.java

```diff
@@ -0,0 +1,320 @@
+            Preconditions.checkArgument(graphName.equals(graphNameOnConfig),
+                                        String.format("Supplied graphName %s does not match property value supplied on config: %s.",
+                                                      graphName,
+                                                      graphNameOnConfig
+                                        ));
+        } else {
+            config.getMap().put(PROPERTY_GRAPH_NAME, graphName);
+        }
+        log.warn("Configuration {} is only guaranteed to take effect when graph {} has been closed and reopened on all Janus Graph Nodes.",
+            graphName,
+            graphName
+        );
+        updateVertexWithProperties(PROPERTY_GRAPH_NAME, graphName, config.getMap());
+    }
+
+    /**
+     * Update Template_Configuration by updating supplied existing properties and adding new ones to the
+ 		 * {@link MapConfiguration}; your updated Configuration may not contain the property "graph.graphname";
+     * NOTE: Any graph using a configuration that was created using the Template_Configuration must--
+     * 1) be closed and reopened on every Janus Graph Node 2) have its corresponding Configuration removed
+     * and 3) recreate the graph-- before the update is guaranteed to take effect.
+     */
+    public void updateTemplateConfiguration(final MapConfiguration config) {
+        Preconditions.checkArgument(!config.containsKey(PROPERTY_GRAPH_NAME),
+                                    String.format("Your updated template configuration may not contain the property \"%s\".",
+                                                  PROPERTY_GRAPH_NAME
+                                    ));
+        log.warn("Any graph configuration created using the Template_Configuration are only guraranteed to have their configuration updated " +
+                 "according to this new Template_Configuration when the graph in question has been closed on every Janus Graph Node, its " +
+                 "corresponding Configuration has been removed, and the graph has been recreated.");
+        updateVertexWithProperties(propertyTemplate, true, config.getMap());
+    }
+
+
+    /**
+     * Remove Configuration according to {@link String graphName}
+     */
+    public void removeConfiguration(final String graphName) {
+        final Vertex v;
+        try {
+            v = graph.traversal().V().has(PROPERTY_GRAPH_NAME, graphName).next();
+        } catch (NoSuchElementException e) {
+            return;
+        }
+        v.remove();
+        graph.tx().commit();
+    }
+
+    /**
+     * Remove Template_Configuration
+     */
+    public void removeTemplateConfiguration() {
+        final Vertex v;
+        try {
+            v = graph.traversal().V().has(propertyTemplate, true).next();
+        } catch (NoSuchElementException e) {
+            return;
+        }
+        v.remove();
+        graph.tx().commit();
```

Comment:
Same comment as above (or maybe add a private method with the implementation and reuse here and above)